lib/web_ui/lib/src/engine/text/line_breaker.dart

```diff
@@ -25,15 +40,109 @@ enum LineBreakType {
 }
 
 /// Splits [text] into fragments based on line breaks.
-class LineBreakFragmenter extends TextFragmenter {
-  const LineBreakFragmenter(super.text);
+abstract class LineBreakFragmenter extends TextFragmenter {
+  factory LineBreakFragmenter(String text) {
+    if (domWindow.Intl.v8BreakIterator != null) {
+      return V8LineBreakFragmenter(text);
+    }
+    return FWLineBreakFragmenter(text);
+  }
+
+  @override
+  List<LineBreakFragment> fragment() {
+    return _computeLineBreakFragments(text);
+  }
+}
+
+/// Flutter web's custom implementation of [LineBreakFragmenter].
+class FWLineBreakFragmenter extends TextFragmenter implements LineBreakFragmenter {
+  FWLineBreakFragmenter(super.text);
 
   @override
   List<LineBreakFragment> fragment() {
     return _computeLineBreakFragments(text);
   }
 }
 
+/// An implementation of [LineBreakFragmenter] that uses Chrome's
+/// `v8BreakIterator` API to find line breaks in the given [text].
+class V8LineBreakFragmenter extends TextFragmenter implements LineBreakFragmenter {
+  V8LineBreakFragmenter(super.text)
+      : assert(domWindow.Intl.v8BreakIterator != null);
+
+  @override
+  List<LineBreakFragment> fragment() {
+    final List<LineBreakFragment> breaks = <LineBreakFragment>[];
+    int fragmentStart = 0;
+
+    final DomV8BreakIterator iterator = createV8BreakIterator();
+
+    iterator.adoptText(text);
+    iterator.first();
+    while (iterator.next() != -1) {
+      final LineBreakType type = _getBreakType(iterator);
+
+      final int fragmentEnd = iterator.current();
+      int trailingNewlines = 0;
+      int trailingSpaces = 0;
+
+      // Calculate trailing newlines and spaces.
+      for (int i = fragmentStart; i < fragmentEnd; i++) {
+        final int codeUnit = text.codeUnitAt(i);
+        if (_kNewlines.contains(codeUnit)) {
```

Comment:
This may be a premature optimization, but is there a way to avoid list iteration in most cases by checking that the code unit is not even in the new line or space character code range?

Reply:
Which iteration do you want to avoid? I'm assuming `Set.contains()` is constant, is it not the case?

Reply:
Didn't realize they were constants, so assumed `.contains` would scan. LGTM.